include/faabric/transport/MpiMessageEndpoint.h

```diff
@@ -6,8 +6,31 @@
 #include <faabric/transport/macros.h>
 
 namespace faabric::transport {
+/* These two abstract methods are used to broadcast the host-rank mapping at
+ * initialisation time.
+ */
 faabric::MpiHostsToRanksMessage recvMpiHostRankMsg();
 
 void sendMpiHostRankMsg(const std::string& hostIn,
                         const faabric::MpiHostsToRanksMessage msg);
+
+/* This class abstracts the notion of a communication channel between two MPI
+ * ranks. There will always be one rank local to this host, and one remote.
+ * Note that the port is unique per (user, function, sendRank, recvRank) tuple.
+ */
+class MpiMessageEndpoint
```

Comment:
It would be nice to add some tests for this class. It contains non-trivial logic and is used quite deep in the stack so it's good to have a simple test that makes debugging issues easier.